src/rest/run_method.ts

```diff
@@ -62,7 +62,11 @@ export async function runMethod<T = any>(
         method,
         reject,
         respond: (data: { status: number; body?: string }) =>
-          resolve(camelize<T>(JSON.parse(data.body || "{}"))),
+          resolve(
+            data.status !== 204
```

Comment:
ideally, you should use the `DiscordHTTPResponseCodes` enum here, specifically, `DiscordHTTPResponseCodes.NoContent` (204).